src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,457 @@
+     * @return True if identAst is used as a left-hand side value otherwise false.
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return isStandAloneIncrementOrDecrement(identAst)
+                || parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is used as
+     * an operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as an operand of stand-alone
+     *         increment or decrement otherwise false
+     */
+    private static boolean isStandAloneIncrementOrDecrement(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        final DetailAST grandParent = parent.getParent();
+        return TokenUtil.isOfType(parent, INCREMENT_AND_DECREMENT_TOKENS)
+                && TokenUtil.isOfType(grandParent, TokenTypes.EXPR)
+                && (!TokenUtil.isOfType(grandParent.getParent(),
+                TokenTypes.ELIST, TokenTypes.INDEX_OP, TokenTypes.ASSIGN)
+                || grandParent.getParent().getParent().getType() == TokenTypes.FOR_ITERATOR);
+    }
+
+    /**
+     * Maintains information about the local variable.
+     */
+    private static final class VariableDesc {
+
+        /**
+         * The name of the variable.
+         */
+        private final String name;
+
+        /**
+         * DetailAst of type {@link TokenTypes#TYPE}.
+         */
+        private final DetailAST typeAst;
+
+        /**
+         * The scope of variable is determined by the DetailAst of type
+         * {@link TokenTypes#SLIST} which is enclosing the variable.
+         */
+        private final DetailAST scope;
+
+        /**
+         * Is the variable used again.
+         */
+        private boolean used;
+
+        /**
+         * Create a new VariableDesc instance.
+         *
+         * @param name name of the variable
+         * @param typeAst DetailAst of type {@link TokenTypes#TYPE}
+         * @param scope {@link VariableDesc#scope}
+         */
+        /* package */ VariableDesc(String name, DetailAST typeAst, DetailAST scope) {
+            this.name = name;
+            this.typeAst = typeAst;
+            this.scope = scope;
+        }
+
+        /**
+         * Get the name of variable.
+         *
+         * @return name of variable
+         */
+        public String getName() {
+            return name;
+        }
+
+        /**
+         * Get the associated {@link VariableDesc#typeAst}.
+         *
+         * @return {@link VariableDesc#typeAst}
+         */
+        public DetailAST getTypeAst() {
+            return typeAst;
+        }
+
+        /**
+         * Get the associated {@link VariableDesc#scope}.
```

Comment:
It would be better to not have to look at private class members to determine what a `scope` is in this context.

Reply:
removed it.

Reply:
removed it

Reply:
I should be more explicit; I think that the language used on private class member "The scope of variable is determined by..." and using the term "enclosing" is very descriptive, but this wording is not used in the constructor or accessor (the "API" of your `VariableDesc` class).

Reply:
Ah, sorry I didn't get it, changed it now.